src/main/java/com/arrl/radiocraft/common/radio/voice/handheld/PlayerRadio.java

```diff
@@ -121,15 +126,36 @@ public void setReceiving(boolean value) {
 
     @Override
     public void transmitAudioPacket(ServerLevel level, short[] rawAudio, int wavelength, int frequencyKiloHertz, UUID sourcePlayer) {
+
+        AntennaNetwork network;
+        synchronized (this) {
+            network = this.network;
+        }
+
         if(network != null) {
             //TODO move transmission logic into the AntennaNetwork
-            Set<IAntenna> antennas = network.allAntennas();
+
+            AntennaPos thisPos;
+            thisPos = getAntennaPos();
+            if(thisPos == null || thisPos.level() == null) return;
+            if(!thisPos.level().equals(level.getServerLevel())) return;
+
+            Set<IAntenna> antennas;
+            Set<IAntenna> masterSet = network.allAntennas();
+
+            synchronized (masterSet) { //you must synchronize on a SynchronizedSet when iterating, making a local copy to avoid locking the list so things like repeaters don't deadlock
+                antennas = new HashSet<>(masterSet);
+            }
 
             for(IAntenna antenna : antennas) {
                 if(antenna != this) {
                     AntennaVoicePacket packet = new AntennaVoicePacket(level, rawAudio.clone(), wavelength, frequencyKiloHertz, 1.0F, this, sourcePlayer);
 
-                    double distance = Math.sqrt(packet.getSource().getBlockPos().distSqr( antenna.getBlockPos()));
+                    AntennaPos pos = antenna.getAntennaPos();
+
+                    if(pos == null) continue;
+
+                    double distance = Math.sqrt(thisPos.position().distSqr(pos.position()));
```

Comment:
Depending on what we use it for, it might make sense to use the squared euclidean distance if we want to save performance.

https://en.wikipedia.org/wiki/Euclidean_distance#Squared_Euclidean_distance

It makes more sense for comparing distance than this calculation.